ipfscluster_test.go

```diff
@@ -1096,6 +1096,41 @@ func TestClustersRecover(t *testing.T) {
 	}
 }
 
+func TestClustersRecoverAll(t *testing.T) {
+	ctx := context.Background()
+	clusters, mock := createClusters(t)
+	defer shutdownClusters(t, clusters, mock)
+	h2 := test.Cid2
+
+	ttlDelay()
+
+	clusters[0].Pin(ctx, h2, api.PinOptions{})
+
+	pinDelay()
+
+	_, err := clusters[rand.Intn(nClusters)].RecoverAll(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	delay()
+
+	info, err := clusters[rand.Intn(nClusters)].Status(ctx, h2)
```

Comment:
The other Recover tests are not a very good example on how to test this as they are old, but at least it trigger the recover pathway.

Here you are pinning a normal Cid and then calling recover which will not do much. But assuming that is more or less "covered" by the other test, you should check that you get the expected global pin info object when recovering an `ErrorCid`.

In short:

```
Pin(cid1)
Pin(errorCid)
RecoverAll(errorCid)
```

Should return a globalPinInfo on which you check:

* Has two items
* len(peermap) == nClusters for each item

The rest is not really necessary, as Status is tested in many other places.